test/e2e/proxy_test.go

```diff
@@ -48,30 +50,41 @@ func (tc *testContext) testCerts(t *testing.T) {
 		certificates.ProxyCertsConfigMap, meta.GetOptions{})
 	require.NoErrorf(t, err, "error getting trusted CA ConfigMap: %w", err)
 
-	// Read all certs from CM data
+	// Read all expected certs from CM data
 	trustedCABundle := cm.Data[certificates.CABundleKey]
 	assert.Greater(t, len(trustedCABundle), 0, "no additional user-provided certs in bundle")
 
-	certs := x509.NewCertPool()
-	require.True(t, certs.AppendCertsFromPEM([]byte(trustedCABundle)), "unable to parse certs from trusted CA ConfigMap data")
-	subjects := certs.Subjects()
-	// Ensure each cert has been imported into every Windows instance's system store
 	for _, node := range gc.allNodes() {
 		t.Run(node.GetName(), func(t *testing.T) {
 			addr, err := controllers.GetAddress(node.Status.Addresses)
 			require.NoError(t, err, "unable to get node address")
 
-			for i, subjectBytes := range subjects {
-				command := fmt.Sprintf("(Get-ChildItem -Path Cert:\\LocalMachine\\Root | "+
-					"Where-Object {$_.Subject -eq '%s'}).Count", string(subjectBytes))
-				out, err := tc.runPowerShellSSHJob(fmt.Sprintf("get-cert-%d", i), command, addr)
+			// Read in one cert at a time and test it exists in the Windows instance's system store
+			scanner := bufio.NewScanner(strings.NewReader(trustedCABundle))
+			scanner.Split(splitAtPEMCert())
+			i := 0
+			for scanner.Scan() {
+				// Multi-line certificate data causes issues in the command. Encode to base64 as a workaround
+				expectedCertBase64 := base64.StdEncoding.EncodeToString([]byte(scanner.Text()))
+				commandToRun := fmt.Sprintf("$base64Data=\\\"%s\\\";"+
+					// Decode base64 into cert's actual string data
+					"$certString=[Text.Encoding]::Utf8.GetString([Convert]::FromBase64String($base64Data));"+
+					// Create a Powershell certificate object with the expected cert.
+					// First requires data to be written to a file and then provide the file path the cert constructor
+					"Set-Content C:\\Temp\\cert.pem $certString;"+
+					"$expectedCert=[System.Security.Cryptography.X509Certificates.X509Certificate2]::new(\\\"C:\\Temp\\cert.pem\\\");"+
```

Comment:
I also tried with passing the cert byte data directly to the `X509Certificate2` constructor and, interestingly enough, reading from a file was consistently speedier.